packages/twenty-server/src/engine/api/graphql/workspace-query-runner/workspace-pre-query-hook/storage/workspace-query-hook.storage.ts

```diff
@@ -0,0 +1,59 @@
+// hook-registry.service.ts
+import { Injectable } from '@nestjs/common';
+import { Module } from '@nestjs/core/injector/module';
+
+import { WorkspaceQueryHookInstance } from 'src/engine/api/graphql/workspace-query-runner/workspace-pre-query-hook/interfaces/workspace-query-hook.interface';
+
+import { WorkspaceQueryHookKey } from 'src/engine/api/graphql/workspace-query-runner/workspace-pre-query-hook/decorators/workspace-query-hook.decorator';
+
+interface WorkspaceQueryHookData<T> {
+  instance: T;
+  host: Module;
+  isRequestScoped: boolean;
+}
+
+@Injectable()
+export class WorkspaceQueryHookStorage {
+  private preHookInstances = new Map<
+    WorkspaceQueryHookKey,
+    WorkspaceQueryHookData<WorkspaceQueryHookInstance>[]
+  >();
+  private postHookInstances = new Map<
+    WorkspaceQueryHookKey,
+    WorkspaceQueryHookData<WorkspaceQueryHookInstance>[]
+  >();
+
+  registerWorkspaceQueryPreHookInstance(
+    key: WorkspaceQueryHookKey,
+    data: WorkspaceQueryHookData<WorkspaceQueryHookInstance>,
+  ) {
+    if (!this.preHookInstances.has(key)) {
+      this.preHookInstances.set(key, []);
+    }
+
+    this.preHookInstances.get(key)?.push(data);
```

Comment:
Does that mean we can register multiple hooks for the same resolver? I'm wondering if at some point we will want to follow a specific order, similarly to middleware (which is kind of what they are). I'm not sure the suggested API will work in this case with decorators and explorer 🤔

Reply:
Yes, as it was possible before, I did the same, so we can have multiple hooks for the same resolver and if later we want to run them into a specific order we can add a property `position` or `index` into the decorator